scripts/pre-commit

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+exit 0
+
+root_dir=$(git rev-parse --show-toplevel)
+cd "$root_dir"
+
+# Read hashes from existing PKGBUILD
+source ./PKGBUILD
+pkgbuild_hashes=(${sha256sums[@]})
+
+# Get actual hashes according to makepkg
+eval $(makepkg -g)
+actual_hashes=(${sha256sums[@]})
+
+# COmpare the hashes
+diff_output=$(diff -u \
+    --label PKGBUILD --label actual \
+    <(printf "%s\n" "${pkgbuild_hashes[@]}") \
+    <(printf "%s\n" "${actual_hashes[@]}"))
+if [[ $? != 0 ]]; then
+    printf "PKGBUILD hashes do not match actual hashes. Diff:\n" >&2
+    printf "%s\n" "$diff_output" >&2
+fi
+
+# Check if arrays are equal. They must be the same length and contain equal
+# elements.
+[[ ${#pkgbuild_hashes[@]} == ${#actual_hashes[@]} ]] || hash_mismatch
```

Comment:
What is this `hash_mismatch`?  Or is this just intentionally to have a "command not found" ?

Also I think a simple:

```bash
[[ "${pkgbuild_hashes[@]}" == "${actual_hashes[@]}" ]] || hash_mismatch
```

should suffice the comparison without actually checking the length and comparing each hash individually.